NetCoreRestApi/DataLayer/Converters/ProductModelConverter.cs

```diff
@@ -1,27 +1,23 @@
-﻿using DataLayer.Models;
-using Common.Converter;
+﻿using Common.Converter;
+using Common.Converters;
 using DataLayer.Entities;
-using System.Collections.Generic;
+using DataLayer.Models;
+using System;
+using System.Linq.Expressions;
 
 namespace DataLayer.Converters
 {
-    public class ProductModelConverter : IConverter<ProductEntity, ProductModel>
+    public class ProductModelConverter<TId> : BaseConverter<ProductEntity<TId>, ProductModel<TId>>
     {
-        private readonly IConverter<CategoryEntity, CategoryModel> _categoryConverter;
-
-        public ProductModelConverter(IConverter<CategoryEntity, CategoryModel> categoryConverter)
+        private readonly IConverter<CategoryEntity<TId>, CategoryModel<TId>> _categoryConverter;
```

Comment:
Remove TId from CategoryEntity and CategoryModel because you specified Tid in BaseEntity and BaseModel classes so

CategoryModel: BaseModel<int>
CategoryEntity: BaseEntity<int>

TId is type int here